platform/osx/os_osx.mm

```diff
@@ -830,62 +832,65 @@ - (void)keyUp:(NSEvent *)event
     _glfwInputKey(window, key, [event keyCode], GLFW_RELEASE, mods);*/
 }
 
+inline void sendScrollEvent (int buttonIndex) {
+    InputEvent ev;
+    ev.type=InputEvent::MOUSE_BUTTON;
+    ev.mouse_button.button_index = buttonIndex;
+    ev.mouse_button.pressed=true;
+    ev.mouse_button.x=mouse_x;
+    ev.mouse_button.y=mouse_y;
+    ev.mouse_button.global_x=mouse_x;
+    ev.mouse_button.global_y=mouse_y;
+    ev.mouse_button.button_mask=button_mask;
+    OS_OSX::singleton->push_input(ev);
+    ev.mouse_button.pressed=false;
+    OS_OSX::singleton->push_input(ev);
+}
+
+void scrollWithMomentum (int x, int y, bool finish) {
+    if (x && fabs(scrollAccumulatorX) >= MOMENTUM_SCROLL_THRESHOLD) {
+        sendScrollEvent(scrollAccumulatorX > 0 ? BUTTON_WHEEL_LEFT : BUTTON_WHEEL_RIGHT);
+        scrollAccumulatorX -= x * MOMENTUM_SCROLL_THRESHOLD;
+        if ((scrollAccumulatorX < 0 && x > 0) || (scrollAccumulatorX > 0 && x < 0)) scrollAccumulatorX = 0;
+    }
+    if (y && fabs(scrollAccumulatorY) >= MOMENTUM_SCROLL_THRESHOLD) {
+        sendScrollEvent(scrollAccumulatorY > 0 ? BUTTON_WHEEL_UP : BUTTON_WHEEL_DOWN);
+        scrollAccumulatorY -= y * MOMENTUM_SCROLL_THRESHOLD;
+        if ((scrollAccumulatorY < 0 && y > 0) || (scrollAccumulatorY > 0 && y < 0)) scrollAccumulatorY = 0;
+    }
+    if (finish) {
+        if (x && scrollAccumulatorX || y && scrollAccumulatorY) scrollWithMomentum(x, y, finish);
+        else scrollAccumulatorX = scrollAccumulatorY = 0;
+    }
+}
+
 - (void)scrollWheel:(NSEvent *)event
 {
-
-	 double deltaX, deltaY;
-
-#if MAC_OS_X_VERSION_MAX_ALLOWED >= 1070
-    if (floor(NSAppKitVersionNumber) > NSAppKitVersionNumber10_6)
-    {
-	deltaX = [event scrollingDeltaX];
-	deltaY = [event scrollingDeltaY];
-
-	if ([event hasPreciseScrollingDeltas])
-	{
-	    deltaX *= 0.1;
-	    deltaY *= 0.1;
-	}
+	double factor = [event hasPreciseScrollingDeltas] ? 1.0 : 0.1,
+	       deltaX = [event deltaX] * factor,
+	       deltaY = [event deltaY] * factor;
+    bool hasMomentum = event.phase != NSEventPhaseNone;
+    bool endScroll = false;
+
+    if (hasMomentum) {
+        switch (event.phase) {
```

Comment:
It is confusing - I chose `phase` over `momentumPhase` because I would getting behavior described by those docs **but** on the `phase` property. Using the scroll wheel on the mouse gives `NSEventPhaseNone` for both properties on every event. In the case of the trackpad `momentumPhase` will give `NSEventPhaseBegan` one event later and then change `NSEventPhaseNone` for subsequent events. Odd behavior